src/main/java/org/dataone/hashstore/filehashstore/FileHashStore.java

```diff
@@ -269,15 +267,13 @@ protected void verifyHashStoreProperties(
      * @throws IOException If `hashstore.yaml` doesn't exist
      */
     protected HashMap<String, Object> loadHashStoreYaml(Path storePath) throws IOException {
-        Path hashstoreYaml = storePath.resolve("hashstore.yaml");
-        File hashStoreYaml = hashstoreYaml.toFile();
+        Path hashStoreYamlPath = storePath.resolve("hashstore.yaml");
+        File hashStoreYamlFile = hashStoreYamlPath.toFile();
         ObjectMapper om = new ObjectMapper(new YAMLFactory());
         HashMap<String, Object> hsProperties = new HashMap<>();
 
         try {
-            HashMap<?, ?> hashStoreYamlProperties = om.readValue(hashStoreYaml, HashMap.class);
-            String yamlStorePath = (String) hashStoreYamlProperties.get("store_path");
-            hsProperties.put(HashStoreProperties.storePath.name(), Paths.get(yamlStorePath));
+            HashMap<?, ?> hashStoreYamlProperties = om.readValue(hashStoreYamlFile, HashMap.class);
             hsProperties.put(
                 HashStoreProperties.storeDepth.name(), hashStoreYamlProperties.get("store_depth")
             );
```

Comment:
in the `writeHashStoreYaml` method, you need to close the writer.

Reply:
Understood, it has been fixed.